app/src/test/java/com/ikarabulut/Game.java

```diff
@@ -0,0 +1,29 @@
+package com.ikarabulut;
+
+public class Game {
+    private Board board;
+
+    public Game(Board theBoard) {
+        board = theBoard;
+    }
+
+    public Board generateNextBoard() {
+        int rows = board.getRows();
+        int columns = board.getColumns();
+        Board nextBoard = new Board(rows, columns);
+
+        for (int row=0; row<rows; row++) {
+            for (int column=0; column<columns; column++) {
+                Cell cell = board.getCell(row, column);
+                int aliveNeighbors = board.getNumberOfAliveNeighbors(row, column);
+                boolean isAliveNextGeneration = Rules.willBeAliveNextGeneration(aliveNeighbors, cell.checkIfAlive());
```

Comment:
@jramnani On this line here. the second parameter being passed `cell.checkIfAlive()` is this a better practice since you are asking the `Cell` object directly, hey are you alive? 

Instead of in my draft I did `board.isCellAliveAt` which is saying hey `Board` object go and ask `Cell` if they are alive here?

Reply:
That's a good question, and to be honest, I don't know. Here are the trade-offs I can think of:

* I think asking a Cell directly is more clear to the reader what's going on.  There's one less level of indirection for the reader to think about.  Also, I think it makes it clear that a Rule is applied to a Cell.
* By getting a Cell from the board, Game depends on Board and Cell. We've increased its dependencies on other objects by one.
* The Cell's API is so small that the probability of a change to Cell breaking the Game object seems low. So the future maintenance cost of adding this dependency is also low.
* If Cell was a larger object with more methods (and a subsequently higher probability of change), then I may consider something like `board.isCellAliveAt` to shield the Game object from a change in Cell rippling out to break Game.
* Board's responsibility is to manage a collection of Cells.  I think its reasonable for collaborators to work on the individual values within the collection.  If a Collection object needed to do a computation across the entire collection: like summing numbers, or counting the total number of something like Alive cells, then it could make sense to tell the Board to do that instead of asking a collaborator to process Cells individually.  But, in Game of Life, we apply the Rules to each Cell individually, based on its individual context (its number of alive neighbors), not across all Cells as an aggregate.

I hope this helps!  Let me know if it spurs any more questions.